hs_core/tests/api/native/test_bag_ingestion.py

```diff
@@ -0,0 +1,114 @@
+# -*- coding: utf-8 -*-
+import os
+import tempfile
+import zipfile
+import shutil
+from unittest import TestCase
+
+from django.contrib.auth.models import Group, User
+from django.core.files.uploadedfile import UploadedFile
+from rdflib import Graph, URIRef
+from rdflib.compare import _squashed_graphs_triples
+from rdflib.namespace import DCTERMS, RDF
+
+from hs_composite_resource.models import CompositeResource
+from hs_core.hydroshare import resource, add_resource_files, current_site_url
+from hs_core.testing import MockIRODSTestCaseMixin
+from hs_core import hydroshare
+from hs_file_types.models import GenericLogicalFile, FileSetLogicalFile, GeoFeatureLogicalFile, GeoRasterLogicalFile, \
+    NetCDFLogicalFile, RefTimeseriesLogicalFile
+
+
+class TestCreateResource(MockIRODSTestCaseMixin, TestCase):
+    def setUp(self):
+        super(TestCreateResource, self).setUp()
+
+        self.tmp_dir = tempfile.mkdtemp()
+        self.hs_group, _ = Group.objects.get_or_create(name='Hydroshare Author')
+        # create a user
+        self.user = hydroshare.create_account(
+            'test_user@email.com',
+            username='mytestuser',
+            first_name='some_first_name',
+            last_name='some_last_name',
+            superuser=False,
+            groups=[self.hs_group]
+        )
+
+        self.test_bag_path = 'hs_core/tests/data/test_resource_metadata_files.zip'
+        self.extracted_directory = "temp"
+
+        with zipfile.ZipFile(self.test_bag_path, 'r') as zip_ref:
+            zip_ref.extractall(self.extracted_directory)
+
+    def tearDown(self):
+        super(TestCreateResource, self).tearDown()
+
+        shutil.rmtree(self.tmp_dir)
+
+        self.user.uaccess.delete()
+        self.user.delete()
+        self.hs_group.delete()
+
+        User.objects.all().delete()
+        Group.objects.all().delete()
+        CompositeResource.objects.all().delete()
```

Comment:
Django does an auto flush of db after running each test, so there is no need to manually delete this db objects. Refer to https://stackoverflow.com/questions/9459030/django-when-to-use-teardown-method#:~:text=In%20general%2C%20django.,Chris%20Pratt%20has%20mentioned%20above. for details.

Reply:
good to know.  I just copied a different test and tried to follow the patterns set there.  I'll try removing the teardown.